src/main/java/com/orange/main/products/service/impl/ProductsServiceImpl.java

```diff
@@ -0,0 +1,77 @@
+package com.orange.main.products.service.impl;
+
+import java.util.List;
+import java.util.Optional;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Sort;
+import org.springframework.stereotype.Service;
+
+import com.orange.main.products.bo.Products;
+import com.orange.main.products.repo.ProductsRepository;
+import com.orange.main.products.service.ProductsService;
+
+import io.micrometer.common.util.StringUtils;
+import jakarta.persistence.EntityManager;
+import jakarta.persistence.PersistenceContext;
+
+@Service
+public class ProductsServiceImpl implements ProductsService {
+
+    @Autowired
+    private ProductsRepository productsRepository;
+
+    @PersistenceContext
+    private EntityManager entityManager;
+
+
+    @Override
+    public Products getProductById(Long productId) {
+        Optional<Products> optionalUser =  productsRepository.findById(productId);
+        return optionalUser.get();
+    }
+
+    @Override
+    public Iterable<Products> getAllProducts() {
+        return productsRepository.findAll();
```

Comment:
What if an error occurs in this case?

Reply:
Becaue we didn't discuss about the details of error message. So I prefer to doing this part after we talk about it.

Reply:
IMHO, any execution related to the Database might cause exceptions, like, DB connection, data mapping... etc, implementing error handling is nothing to do with showing error messages (which we could discuss later) in the frontend, or maybe implementing error handling at the next commit in this PR?